src/Administration/Login/StateValidator.php

```diff
@@ -0,0 +1,63 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Administration\Login;
+
+use Shopware\Administration\Login\Exception\LoginException;
+use Shopware\Core\Framework\Log\Package;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\String\ByteString;
+use Symfony\Component\Validator\Constraints\EqualTo;
+use Symfony\Component\Validator\Constraints\Length;
+use Symfony\Component\Validator\Constraints\NotBlank;
+use Symfony\Component\Validator\Constraints\NotNull;
+use Symfony\Component\Validator\Validation;
+
+/**
+ * @internal
+ */
+#[Package('after-sales')]
+final class StateValidator
+{
+    final public const SESSION_KEY = 'sw_sso_session_key';
+
+    public const RANDOM_LENGTH = 64;
+
+    public function validateRequest(Request $request): void
+    {
+        $this->validateState(
+            $request->get('rdm'),
+            $request->getSession()->get(self::SESSION_KEY),
+        );
+
+        $request->request->set('grant_type', 'shopware_grant');
+        $request->request->set('code', $request->get('code'));
+    }
+
+    public function createRandom(Request $request): string
+    {
+        $random = ByteString::fromRandom(self::RANDOM_LENGTH)->toString();
+
+        $request->getSession()->set(self::SESSION_KEY, $random);
+
+        return $random;
+    }
+
+    private function validateState(?string $state, ?string $storedState): void
```

Comment:
honestly is there a reason for using the validator here? actually what you do is compare if the two strings are equal and check the length is 64 characters 

honestly it took me quite a while to figure that out reading the validator code here

Reply:
I'd be glad to have a validator as our Admin Login State has multiple reasons why we have been redirected (login, link or "password confirm"). If this is something I could hook into it does not sound that bad to me

Reply:
I see no evil here. I actually find it easy to read too 

First it checks for a valid generated random and then if it equals the stored